mobile/components/views/Attractions/Search/api/attractionsApi.ts

```diff
@@ -0,0 +1,27 @@
+import { Attraction } from "../../types";
+
+export const fetchUsersAttractions = (
+  id: number
+): Promise<Attraction[]> => {
+  const url = `http://localhost/user/${id}/atrractions`;
+  return fetch(url)
+    .then((response) => response.json())
+    .then((data) => data as Attraction[])
```

Comment:
You need to add `Attraction[] | null | ` Because you dont have 100% quarantee that respone will always retrieve `Attraction[]`

Reply:
Here I followed your code XD

Reply:
So you can adjust also my code or point where it had to be adjusted

Reply:
My bad, no need to add null type, if there is no attraction with id we fetch, we shouldnt get 200 status so `catch` will catch it.

Reply:
If the fetch function returns something other than the expected attraction array, this will be handled in the .then((data) => data as Attraction[]) block. In this case, the as operator is used to cast data to the Attraction[] type. If the date cannot be converted to the expected type, JavaScript automatically throws an exception, which in turn catches the .catch((error) => {...}) block.

Reply:
I mentioned about it in previous comment.

Reply:
Also it not works like you said. The .catch((error) => {...}) block in a Promise chain is triggered by a runtime error, such as an exception thrown manually or by some operations failing, not by type mismatches.